### DISCUSSION TITLE
Siunitx and Significance Stars

### DISCUSSION THREAD

**Discussion by jaaschauer** [lvjr/tabularray#218] [Q&A]
When using siunitx in tabularray, the significance stars are displayed before numbers. I'm using siunitx v2 since i can't get 'add-integer-zero = false' in v3 (and it's required in APA7 style; if you have possible alternatives to get that, please let me know).

```latex
\documentclass{article}

\usepackage{xcolor}
\usepackage{tabularray}
\UseTblrLibrary{siunitx}
\newcommand\SiGuard[1]{{#1}}
\sisetup{detect-all, math-rm=\mathsf, table-auto-round = true}

\begin{document}
\begin{tblr}{
colspec = {
	l
	Q[c, m, si = {table-format = -1.2}]
	Q[c, m, si = {table-format = >0.3, add-integer-zero = false, table-space-text-post = \TblrNote{***}}]
},
row{1} = {cmd=\SiGuard}
	}
	
  Alpha   & Beta  & Gamma  \\  
\hline
  Alpha   & 1.35  & .043{{\TblrNote{*}}} \\
\hline
  Epsilon & -2.34 & <.001{{\TblrNote{***}}} \\
\hline
  Iota    & 2.45 & .002{{\TblrNote{**}}} \\
\hline
 \end{tblr}
\end{document}
```

<img width="293" alt="Bildschirmfoto 2022-03-25 um 11 28 04" src="https://user-images.githubusercontent.com/102300716/160104007-e87e10eb-fb0a-43c4-b1df-17cf702e07b7.png">

I get the error "! Missing number, treated as zero.".  
When using the v3 siunitx command to reserve space for the notes via 

```latex
Q[c, m, si = {table-format = >0.3{\TblrNote{***}}, add-integer-zero = false}]
```

it produces this together with many many error messages:

<img width="293" alt="Bildschirmfoto 2022-03-25 um 11 27 11" src="https://user-images.githubusercontent.com/102300716/160104512-25d2c5a6-003a-4ab3-b4a7-ebcf6ab5c05c.png">

Guess that tabularray is made to be used with siunitx v3. However, is there a way to get this running somehow? Using "add-integer-zero" exclusively works fine.

I also tested with siunitx v3 in overleaf ommitting the 'add-integer-zero' to test what that does. It's better, but again not the desired output since alignment and displaying the comparator sign aren't as expected:

<img width="665" alt="Bildschirmfoto 2022-03-25 um 11 38 02" src="https://user-images.githubusercontent.com/102300716/160105951-7618ae50-1970-4e55-bd4a-8c0022bcd396.png">

**Comment by lvjr** [comment]
Sorry, I only know little about `siunitx`. You may ask this question on https://tex.stackexchange.com

**Reply by jaaschauer** [reply]
Allright, i'll do that. Maybe @josephwright knows something?

**Reply by josephwright** [reply]
I've just tried your example, and with current release `siunitx` plus adding `\usepackage[T1]{fontenc}` (which is really 'required' for any real document), it looks fine. I'd need a log, etc., to help, but that would be best done in an issue in the `siunitx` tracker, and using it directly not via `tabularray`.

**Reply by jaaschauer** [reply]
Thank you! No worries, i get along now, just found the reason:

It's the `\sisetup{table-auto-round = true}` causing trouble. Without it's all fine. So whenever `table-space-text-post` is specified for a Q-column, `table-auto-round = true` should not be used in these columns. Also space is only reserved correctly when using `Q[c, m, si = {table-space-text-post = \textsuperscript{***}}]` instead of `\TblrNote{***}`. However, both restrictions only holds true for `siunitx` v2.

I also did some tests with `siunitx` v3, where some problems persist and also some new ones appear. Here, it's rather the `\sisetup{detect-all}` causing problems (Note: I added the integer zero to numbers here since it's necessary in v3):

<img width="203" alt="Bildschirmfoto 2022-03-26 um 11 00 49" src="https://user-images.githubusercontent.com/102300716/160234624-b774b4bb-5f02-4a0c-a3e6-3f7e1b825026.png">

Even when i do not use `\sisetup` at all, it won't work perfectly:

<img width="183" alt="Bildschirmfoto 2022-03-26 um 10 48 27" src="https://user-images.githubusercontent.com/102300716/160234145-945e1e6c-2622-4dbf-8cf7-99a076cc18dc.png">

Maybe it would be good to improve the interplay of siunitx and tabularray, both are great and probably part of the latex future! 

So should i make a new issue with this in the `siunitx` tracker @josephwright? You say it worked with v3 for you?

And something i'd really appreciate is to have `add-integer-zero = false` again in `siunitx` v3 :)

I used this code for the last example:

```latex
\documentclass{article}

\usepackage[T1]{fontenc}
\usepackage{xcolor}
\usepackage{tabularray}
\UseTblrLibrary{siunitx}

% This causes problems
%\sisetup{math-rm=\mathsf}

\begin{document}

\begin{tblr}{
colspec = {
	l
	Q[c, m, si = {table-format = -1.2}]
	Q[c, m, si = {table-format = <0.3{\TblrNote{*}}}]}
	}
	
  Alpha   & {{{Beta}}}  & {{{Gamma}}}  \\  
\hline
  Alpha   & 1.35  & 0.043\TblrNote{*} \\
\hline
  Epsilon & -2.34 & <0.001\TblrNote{***} \\
\hline
  Iota    & 2.45 & 0.002\TblrNote{**} \\
\hline
\end{tblr}
\end{document}
```

**Reply by josephwright** [reply]
I think you are not using the latest release: like I said, please post to the `siunitx` repo with an example using the package directly

**Reply by jaaschauer** [reply]
@josephwright here the log file of this last example, if needed:
[output.log](https://github.com/lvjr/tabularray/files/8355141/output.log)

And the log of the v2 problem just in case you want to have a look:
[test.log](https://github.com/lvjr/tabularray/files/8355144/test.log)
But probably doesn't make much sense to further investigate the v2 problem since at least i found some way to get it going and i guess no one really uses it anymore (unless using APA style).

**Reply by jaaschauer** [reply]
I did this test via overleaf, so maybe it't not the latest release. Allgood, so it seems to be kind of solved :)
